Sources/Sentry/SentryUIViewControllerPerformanceTracker.m

```diff
@@ -109,6 +120,55 @@ - (void)createTransaction:(UIViewController *)controller
     }
 }
 
+- (void)createTimeToDisplay:(UIViewController *)controller
+{
+    SentrySpan *vcSpan = [self viewControllerPerformanceSpan:controller];
+
+    if (![vcSpan isKindOfClass:[SentryTracer self]]) {
+        // Since TTID and TTFD are meant to the whole screen
+        // we will not track child view controllers
+        return;
+    }
+
+    if ([vcSpan.tracer getMiddlewaresOfType:SentryTimeToDisplayTracker.self].count > 0) {
+        // Already tracking time to display.
+        return;
+    }
+
+    if (self.currentTTDTracker) {
+        [self.currentTTDTracker stopWaitingFullDisplay];
+    }
+
+    SentryTimeToDisplayTracker *ttdTracker =
+        [[SentryTimeToDisplayTracker alloc] initForController:controller
+                                           waitForFullDisplay:self.enableWaitForFullDisplay];
+    [vcSpan.tracer addMiddleware:ttdTracker];
+
+    self.currentTTDTracker = ttdTracker;
+}
+
+- (void)finishTTIDForController:(UIViewController *)controller
+{
+    SentrySpan *vcSpan = [self viewControllerPerformanceSpan:controller];
+
+    if (![vcSpan isKindOfClass:[SentryTracer self]]) {
+        // Only tracer may have TimeToDisplayTracker
+        return;
+    }
+
+    SentryTimeToDisplayTracker *ttdTracker =
+        [vcSpan.tracer getMiddlewaresOfType:SentryTimeToDisplayTracker.self].firstObject;
+    [ttdTracker registerInitialDisplay];
+}
+
+- (void)reportFullyDisplayed
+{
+    [self.currentTTDTracker registerFullDisplay];
+
+    // No need for it anymore
+    self.currentTTDTracker = nil;
+}
```

Comment:
`h`: If I've not mistaken, only the hub calls this method. Don't we also have to set `self.currentTTDTracker` to `nil` when the user doesn't manually call the API?

Reply:
This is not required anymore. `currentTTDTracker` is a weak reference and will take care of itself